src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -543,5 +575,117 @@ private async Task DeleteBatchFiles(Guid releaseId, File dataFile)
                 );
             });
         }
+
+        private async Task<Guid> CreateReleaseAndSubjectHierarchy(Release release, string subjectFilename,
+            string subjectName)
+        {
+            if (release.Publication.Topic.Theme == null)
```

Comment:
Rather than have a check like this, it'd probably be a good idea just to perform a new query for the entire relationship. I think if the relationship entities have been previously loaded, the EF context will just pull the entities out of its cache and there shouldn't be another SQL call.